src/hotspot/share/cds/dumpTimeClassInfo.cpp

```diff
@@ -73,12 +77,26 @@ void DumpTimeClassInfo::add_verification_constraint(InstanceKlass* k, Symbol* na
 
   if (log_is_enabled(Trace, aot, verification)) {
     ResourceMark rm;
-    log_trace(aot, verification)("add_verification_constraint: %s: %s must be subclass of %s [0x%x] array len %d flags len %d",
-                                 k->external_name(), from_name->as_klass_external_name(),
-                                 name->as_klass_external_name(), c, vc_array->length(), vcflags_array->length());
+    log_trace(aot, verification)("add_verification_constraint: %s: %s must be subclass of %s [0x%x]",
+                                 _klass->external_name(), from_name->as_klass_external_name(),
+                                 name->as_klass_external_name(), c);
   }
 }
 
+void DumpTimeClassInfo::add_old_verification_dependency(Symbol* name) {
+  if (_old_verifier_dependencies == nullptr) {
+    _old_verifier_dependencies = new (mtClass) GrowableArray<Symbol*>(4, mtClass);
+  }
+  _old_verifier_dependencies->append(name);
```

Comment:
I wonder why there's append and push that do the same thing in GrowableArray.  I'm used to seeing push().

Reply:
I am not sure why there are two versions.  They seem to be used at similar frequency (about 300 calls each).